Core/src/org/sleuthkit/autopsy/datasourcesummary/datamodel/DataSourceUserActivitySummary.java

```diff
@@ -0,0 +1,439 @@
+        assertValidCount(count);
+
+        List<TopWebSearchResult> results = caseProvider.get().getBlackboard().getArtifacts(ARTIFACT_TYPE.TSK_WEB_SEARCH_QUERY.getTypeID(), dataSource.getId())
+                .stream()
+                // get items where search string and date is not null
+                .map(artifact -> {
+                    String searchString = DataSourceInfoUtilities.getStringOrNull(artifact, TYPE_TEXT);
+                    Date dateAccessed = DataSourceInfoUtilities.getDateOrNull(artifact, TYPE_DATETIME_ACCESSED);
+                    return (StringUtils.isNotBlank(searchString) && dateAccessed != null)
+                            ? new TopWebSearchResult(searchString, dateAccessed)
+                            : null;
+                })
+                // remove null records
+                .filter(result -> result != null)
+                // get these messages grouped by search to string
+                .collect(Collectors.groupingBy((result) -> result.getSearchString().toUpperCase()))
+                .entrySet()
+                .stream()
+                // get the most recent access per account type
+                .map((entry) -> entry.getValue().stream().max(TOP_WEBSEARCH_RESULT_DATE_COMPARE).get())
+                // get most recent accounts accessed
+                .sorted(TOP_WEBSEARCH_RESULT_DATE_COMPARE.reversed())
+                .limit(count)
+                // get as list
+                .collect(Collectors.toList());
+
+        // get translation if possible
+        if (translationService.hasProvider()) {
+            for (TopWebSearchResult result : results) {
+                result.setTranslatedResult(getTranslationOrNull(result.getSearchString()));
+            }
+        }
+
+        return results;
+    }
+
+    /**
+     * Return the translation of the original text if possible and differs from
+     * the original. Otherwise, return null.
+     *
+     * @param original The original text.
+     *
+     * @return The translated text or null if no translation can be determined
+     *         or exists.
+     */
+    private String getTranslationOrNull(String original) {
+        if (!translationService.hasProvider() || StringUtils.isBlank(original)) {
+            return null;
+        }
+
+        String translated = null;
+        try {
+            translated = translationService.translate(original);
+        } catch (NoServiceProviderException | TranslationException ex) {
+            logger.log(Level.WARNING, String.format("There was an error translating text: '%s'", original), ex);
+        }
+
+        // if there is no translation or the translation is the same as the original, return null.
+        if (StringUtils.isBlank(translated)
+                || translated.toUpperCase().trim().equals(original.toUpperCase().trim())) {
```

Comment:
There is a warning about dereferencing a null pointer.

Reply:
I think the `StringUtils.isBlank` checks on the `original` variable and the `translated` variable prevent either of these from being null.